### ISSUE TITLE
Build fails if zlib is not installed in default location, despite being available through pkg-config

### ISSUE THREAD

**Issue by Tyg13** [alexcrichton/ssh2-rs#169]
As reported in [rust-lang#69552](https://github.com/rust-lang/rust/issues/69552), building this crate fails if `zlib.h` is not available on search path for the invoked compiler. This can obviously be worked around by setting `DEP_Z_INCLUDE` to the path containing said file, but (as the opener of that issue mentioned), the required include path would be available via `pkg-config`.

A few things:
1. It might be useful to check `pkg-config` in the case that `DEP_Z_INCLUDE` is unset, to avoid this issue in the future.
2. Better error reporting might be helpful, for when this crate is compiled as a dependency. I'm not overly familiar with how `build.rs` scripts work, but perhaps a test compile with an error message saying "Hey set `DEP_Z_INCLUDE` so libssh2-sys can find `zlib.h`" would be helpful.

**Comment by wez** [comment]
This is the sort of thing that is easiest to fix by someone with a system that reproduces the issue so that there is no guesswork or back-and-forth.  I don't see any information about that environment either here or in the other issue and I don't have a ton of time to sink into this, so I'm looking to you (or really @MagicalTux) to help move things forward.  At a minimum I'd like to know if there is say a docker image that has this unusual configuration that we could put into our CI, but I'd also love to accept PR :)

**Comment by MagicalTux** [comment]
@wez as suggested I made this available through docker. Note that this is extremely experimental.

In order to run a shell (the options are there in order to make access to fuse possible):

    docker run --rm -it --privileged --cap-add SYS_ADMIN --cap-add MKNOD azusaos/azusa:latest /bin/bash -i

Might feel a bit slow as there are some optimizations pending, and this is still highly experimental.

**Comment by MagicalTux** [comment]
Note that I never coded anything in rust in my life until today.

```patch
diff --git a/libssh2-sys/build.rs b/libssh2-sys/build.rs
index 1a406a3..4dcaec1 100644
--- a/libssh2-sys/build.rs
+++ b/libssh2-sys/build.rs
@@ -123,6 +123,11 @@ fn main() {
     if let Some(path) = env::var_os("DEP_Z_INCLUDE") {
         cfg.include(path);
     }
+    if let Ok(lib) = pkg_config::find_library("zlib") {
+        for path in &lib.include_paths {
+            cfg.include(path);
+        }
+    }
 
     if let Some(path) = env::var_os("DEP_OPENSSL_INCLUDE") {
         if let Some(path) = env::split_paths(&path).next() {
```

Result:
```
bash-5.0# cargo build
   Compiling libssh2-sys v0.2.15 (/root/ssh2-rs/libssh2-sys)
   Compiling ssh2 v0.8.0 (/root/ssh2-rs)
    Finished dev [unoptimized + debuginfo] target(s) in 20.35s
```

**Comment by wez** [comment]
Published libssh2-sys v0.2.16 with this fix.  Thanks!
